social.md

```diff
@@ -11,134 +11,273 @@ support:
+All of the integrations, regardless of language, should have a consistent look and feel for the rendered buttons for social providers.
 
-## Implementing Popup-Based Workflows
+**Recommendation:**
 
-Single-page applications will need to query the server to determine which social
-providers are configured, and which fields to render for the login or
-registration views.  In either context, the front-end application should make
-a GET request to the login or registration endpoint, to fetch the view model
-as JSON.  The view model will contain the necessary information.  See [login][]
-and [registration][] pages for the view model definitions.
+The [Social Buttons for Bootstrap](https://lipis.github.io/bootstrap-social/) has a very clean consistent look and feel and has been adopted into the Java SDK. Cons for this approach are a dependency on Bootstrap.
 
-## Access Tokens and Authorization Codes
+![social buttons](images/socialbuttons.png)
 
-This is a reference to supported functionality from both Stormpath and different social providers with their OAuth flows. 
+**Pros:** Consistent, clean look and feel across all of Stormpath's integrations.
 
-**Stormpath REST API Support**
+**Cons:** Depends on Bootstrap
 
-|Social Provider|Authorization Code|Access Token|
-|---|---|---|
-|Facebook|NO|YES|
-|Google|YES|YES|
-|LinkedIn|YES|YES|
-|GitHub|NO|YES|
+<a name="page-based"></a>
+### Page-based Redirect Workflow
 
-**Social Provider OAuth Support**
+In this situation, the user leaves your login page by redirect and is taken to
+the provider for authentication.  Once authentication is complete, the user is
+redirected back to a "callback" URI on your website.  This callback will provide
+an access token or access code as a query parameter.  Your server uses a
+confidential keypair of the provider to validate the token/code, then retrieves
+the account data of the authenticated user.
 
-|Social Provider|Authorization Code|Implicit|
-|---|---|---|
-|Facebook|YES|YES|
-|Google|YES|NO|
-|LinkedIn|YES|NO|
-|GitHub|YES|NO|
+<a name="ajax-based"></a>
+### AJAX-based Workflow
 
-Implementation of social login is up to the framework integration, but here are some potential ways to approach it: 
+In this situation the user does not leave your login page.  Instead a popup
+window is created, and the user authenticates with the provider in that window.
+The popup window is created by a JavaScript API in the browser, and when the
+user is done with the popup window (either by authenticating or rejecting the
+prompt) the JavaScript API will call back to your JavaScript application.  If
+the user has authenticated you will be provided with a token/code, which you
+must send to your server and validate with your confidential provider
+credentials.  At this point the workflow is identical to the page-base redirect
```

Comment:
Depends on how we're doing auth. If we're doing cookie auth, we can still do the page based redirect flow. If we're doing token via headers, we'll need to use the popup flow, so the SPA can get the token, or support the implicit grant type ([see comment here](https://github.com/stormpath/stormpath-framework-spec/pull/132#issuecomment-264520008))